daisy_workflows/image_build/windows_for_containers/container_image_install.ps1

```diff
@@ -47,61 +47,163 @@ function Get-MetadataValue {
+  if ($windows_version -eq '2019') {
+    return 'ltsc2019'
+  }
+  # '1709', '1803', '1809':
+  return $windows_version
+}
 
-    $docker_version = "18.09"
-    Write-Host "Installing Docker EE ${docker_version}"
-    Install-Package -Name docker -ProviderName DockerMsftProvider -Force -RequiredVersion ${docker_version}
+function Supports-NanoserverContainerImage {
+  param (
+    [parameter(Mandatory=$true)]
+      [string]$windows_version
+  )
 
-    Write-Host 'Enabling IPv6'
-    $ipv_path = 'HKLM:\SYSTEM\CurrentControlSet\services\TCPIP6\Parameters'
-    Set-ItemProperty -Path $ipv_path -Name 'DisabledComponents' -Value 0x0
+  # Windows Server 2019 does not support the nanoserver container image, only
+  # Servercore:
+  # https://blogs.technet.microsoft.com/virtualization/2018/11/13/windows-server-2019-now-available/.
+  if ($windows_version -eq '2019') {
+    return $false
+  }
+  return $true
+}
 
-    Write-Host 'Restarting computer to finish install'
-    Restart-Computer -Force
+function Get-ServerCoreImageName {
+  param (
+    [parameter(Mandatory=$true)]
+      [string]$windows_version
+  )
+
+  $repo = Get-ContainerRepo $windows_version
+  if ($windows_version -eq '1709' -or $windows_version -eq '1803') {
+    $image = 'windowsservercore'
+  } else {
+    $image = 'servercore'
   }
-  else {
-    # For some reason the docker service may not be started automatically on the
-    # first reboot, although it seems to work fine on subsequent reboots. The
-    # docker service must be running or else the vEthernet interface may not be
-    # present.
-    Restart-Service docker
-
-    Write-Host 'Setting host vEthernet MTU to 1460'
-    Get-NetAdapter | Where-Object {$_.Name -like 'vEthernet*'} | ForEach-Object {
-      & netsh interface ipv4 set subinterface $_.InterfaceIndex mtu=1460 store=persistent
-    }
+  $label = Get-ContainerVersionLabel $windows_version
+  return "${repo}/${image}:${label}"
+}
 
-    # As most if not all Windows containers are based on one of these images
-    # we pull it here so that running a container using this image is quick.
-    Write-Host 'Pulling latest Windows containers'
-    & docker pull "microsoft/windowsservercore:${windows_version}"
-    if (!$?) {
-      throw "Error running 'docker pull microsoft/windowsservercore:${windows_version}'"
-    }
-    & docker pull "microsoft/nanoserver:${windows_version}"
-    if (!$?) {
-      throw "Error running 'docker pull microsoft/nanoserver:${windows_version}'"
-    }
+function Get-NanoserverImageName {
+  param (
+    [parameter(Mandatory=$true)]
+      [string]$windows_version
+  )
+  $repo = Get-ContainerRepo $windows_version
+  $image = 'nanoserver'
+  $label = Get-ContainerVersionLabel $windows_version
+  return "${repo}/${image}:${label}"
+}
+
+function Get-BaseContainerImageNames {
+  param (
+    [parameter(Mandatory=$true)]
+      [string]$windows_version
+  )
+  $images = @(Get-ServerCoreImageName $windows_version)
```

Comment:
What about mcr.microsoft.com/windows:1809, is this a new container?

Reply:
I believe that's the full-blown Windows container that supports every Windows API. Microsoft announced this at Build earlier this year and is making it available with 1809 and 2019. I don't know exactly how big it is but it's probably huge. I don't think we need to add it to our for-containers images at this time but let me know if you think we should.